sql/hive/src/main/scala/org/apache/spark/sql/hive/client/HiveClientImpl.scala

```diff
@@ -1092,14 +1092,23 @@ private[hive] object HiveClientImpl extends Logging {
       hiveTable.setViewExpandedText(t)
     }
 
+    // hive may convert schema into lower cases while bucketSpec will not
+    // only convert if case not match
+    def restoreHiveBucketSpecColNames(schema: StructType, names: Seq[String]): Seq[String] = {
+      names.map { name =>
+        schema.find(col => SQLConf.get.resolver(col.name, name)).map(_.name).getOrElse(name)
+      }
+    }
+
     table.bucketSpec match {
       case Some(bucketSpec) if !HiveExternalCatalog.isDatasourceTable(table) =>
         hiveTable.setNumBuckets(bucketSpec.numBuckets)
-        hiveTable.setBucketCols(bucketSpec.bucketColumnNames.toList.asJava)
```

Comment:
To clarify: `hiveTable.setFields` lower-cases the column names, but `hiveTable.setBucketCols` does not. And this causes the exception?

Reply:
yes